src/ripple/app/main/Main.cpp

```diff
@@ -394,6 +395,11 @@ int run (int argc, char** argv)
         try
         {
             config->VALIDATION_QUORUM = vm["quorum"].as <std::size_t> ();
+            if (vm["quorum"].as <std::size_t>() < 1)
+            {
+                throw std::invalid_argument(
+                        std::to_string(vm["quorum"].as <std::size_t>()));
+            }
```

Comment:
Your analysis is spot-on: `boost::any` is a bit weird you can't get the underlying string; only the converted integer, if there is one.

My personal preference would have been to make those values accept a `std::string`, and then use `boost::lexical_cast`, but I doubt it's worth making this change at this point. Instead let's just simply do what you suggest: simply `throw std::domain_error("")` from here.

Then we change the handler inside the `catch` to say simply:

    std::cerr << "Invalid value specified for --quorum" << std::endl;

The same fix ought to also be made in the `catch` handler for `--rpc_port`.